bbq/compiler/extended_elaboration.go

```diff
@@ -318,3 +352,17 @@ func (e *ExtendedElaboration) FunctionExpressionFunctionType(expression *ast.Fun
 func (e *ExtendedElaboration) IndexExpressionTypes(expression *ast.IndexExpression) (types sema.IndexExpressionTypes, contains bool) {
 	return e.elaboration.IndexExpressionTypes(expression)
 }
+
+func (e *ExtendedElaboration) InterfaceTypeDeclaration(interfaceType *sema.InterfaceType) *ast.InterfaceDeclaration {
+	return e.elaboration.InterfaceTypeDeclaration(interfaceType)
+}
+
+// Elaboration returns the underlying elaboration.
+// IMPORTANT: Only use the original elaboration for type-checking use-cases.
+// It is safe to use in type-checker, since the type checker doesn't rely on
+// extra information added by the desugar phase.
+// Never use it in the compiler or desugar, as it may not include the
+// type information added during the desugar phase.
+func (e *ExtendedElaboration) Elaboration() *sema.Elaboration {
```

Comment:
Thanks for adding a great comment! Maybe "extended elaboration" can be renamed to something like "desugared elaboration", an elaboration of a desugared program, to make this more obvious

Reply:
Renamed in https://github.com/onflow/cadence/pull/3886/commits/44541b68ac26c329a804a59533ec706182097509